src/main/java/org/jabref/logic/openoffice/backend/NamedRangeReferenceMark.java

```diff
@@ -0,0 +1,453 @@
+        cursor.goRight((short) n, true); // select the newly inserted spaces
+        return cursor;
+    }
+
+    private static void createReprInDocument(XTextDocument doc,
+                                             String refMarkName,
+                                             XTextCursor position,
+                                             boolean insertSpaceAfter,
+                                             boolean withoutBrackets)
+        throws
+        CreationException {
+
+        // The cursor we received: we push it before us.
+        position.collapseToEnd();
+
+        XTextCursor cursor = safeInsertSpacesBetweenReferenceMarks(position.getEnd(), 2);
+
+        // cursors before the first and after the last space
+        XTextCursor cursorBefore = cursor.getText().createTextCursorByRange(cursor.getStart());
+        XTextCursor cursorAfter = cursor.getText().createTextCursorByRange(cursor.getEnd());
+
+        cursor.collapseToStart();
+        cursor.goRight((short) 1, false);
+        // now we are between two spaces
+
+        final String left = NamedRangeReferenceMark.REFERENCE_MARK_LEFT_BRACKET;
+        final String right = NamedRangeReferenceMark.REFERENCE_MARK_RIGHT_BRACKET;
+        final short leftLength = (short) left.length();
+        final short rightLength = (short) right.length();
+        String bracketedContent = (withoutBrackets
+                                   ? ""
+                                   : left + right);
+
+        cursor.getText().insertString(cursor, bracketedContent, true);
+
+        UnoReferenceMark.create(doc, refMarkName, cursor, true /* absorb */);
+
+        cursorBefore.goRight((short) 1, true);
+        cursorBefore.setString("");
+        if (!insertSpaceAfter) {
+            cursorAfter.goLeft((short) 1, true);
+            cursorAfter.setString("");
+        }
+    }
+
+    static NamedRangeReferenceMark create(XTextDocument doc,
+                                          String refMarkName,
+                                          XTextCursor position,
+                                          boolean insertSpaceAfter,
+                                          boolean withoutBrackets)
+        throws
+        CreationException {
+
+        createReprInDocument(doc, refMarkName, position, insertSpaceAfter, withoutBrackets);
+        return new NamedRangeReferenceMark(refMarkName);
+    }
+
+    /**
+     * @return Optional.empty if there is no corresponding range.
+     */
+    static Optional<NamedRangeReferenceMark> getFromDocument(XTextDocument doc, String refMarkName)
+        throws
+        NoDocumentException,
+        WrappedTargetException {
+        return (UnoReferenceMark.getAnchor(doc, refMarkName)
+                .map(e -> new NamedRangeReferenceMark(refMarkName)));
+    }
+
+    /*
+     * Remove it from the document.
+     *
+     * See: removeCitationGroups
+     */
+    @Override
+    public void nrRemoveFromDocument(XTextDocument doc)
```

Comment:
Same here: Please avoid repeating the namespace / class in the method title.

Reply:
ok.